From 66316e7efc3d00460e4abc1c2cc91172e2c34087 Mon Sep 17 00:00:00 2001
From: Nicolas LOUBOUTIN <nicolas.louboutin@st.com>
Date: Mon, 3 Nov 2025 08:25:52 +0100
Subject: [PATCH 14/14] board: st: set density level depending on detected
 display

HDMI : set density parameter to 320
LVDS : set density parameter to 240

Signed-off-by: Nicolas LOUBOUTIN <nicolas.louboutin@st.com>
---
 board/st/stm32mp2/stm32mp2.c       | 11 +++++++++--
 include/configs/stm32mp25_common.h |  1 +
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/board/st/stm32mp2/stm32mp2.c b/board/st/stm32mp2/stm32mp2.c
index 5bce1999cd6..29154f297d2 100644
--- a/board/st/stm32mp2/stm32mp2.c
+++ b/board/st/stm32mp2/stm32mp2.c
@@ -318,12 +318,15 @@ static void board_stm32mp25x_eval_init(void)
 	/* auto detection of connected hdmi bridge */
 	compatible = detect_device(stm32mp25x_bridges, ARRAY_SIZE(stm32mp25x_bridges));
 
-	if (!compatible)
+	if (!compatible) {
 		/* remove the hdmi bridge in environment */
 		env_set("hdmi", "");
-	else
+		env_set("android_lcd_bootargs", "lcd_density=240");
+	} else {
 		/* save the detected compatible in environment */
 		env_set("hdmi", compatible);
+		env_set("android_lcd_bootargs", "lcd_density=320");
+	}
 }
 
 static void board_stm32mp25x_disco_init(void)
@@ -338,6 +341,7 @@ static void board_stm32mp25x_disco_init(void)
 	if (!compatible) {
 		/* remove the panel in environment */
 		env_set("panel", "");
+		env_set("android_lcd_bootargs", "lcd_density=320");
 
 		/* no panel detected then unbind lvds to avoid a bad clock tree */
 		node = ofnode_by_compatible(ofnode_null(), "st,stm32mp25-lvds");
@@ -350,6 +354,7 @@ static void board_stm32mp25x_disco_init(void)
 	} else {
 		/* save the detected compatible in environment */
 		env_set("panel", compatible);
+		env_set("android_lcd_bootargs", "lcd_density=240");
 	}
 }
 
@@ -697,6 +702,7 @@ static int fixup_stm32mp257_eval_panel(void *blob)
 		/* string compare the hdmi compatible limit to 10 chars (adi,adv753) */
 		detect_adv753x = !strncmp(hdmi, "adi,adv753x", 10);
 
+
 	/* update LVDS panel "edt,etml0700z9ndha" */
 	status = detect_etml0700z9ndha ? FDT_STATUS_OKAY : FDT_STATUS_DISABLED;
 	nodeoff = fdt_set_status_by_compatible(blob, "edt,etml0700z9ndha", status);
@@ -754,6 +760,7 @@ static int fixup_stm32mp257_disco_panel(void *blob)
 	if (panel)
 		detect_etml0700z9ndha = !strcmp(panel, "edt,etml0700z9ndha");
 
+
 	/* update LVDS panel "edt,etml0700z9ndha" */
 	status = detect_etml0700z9ndha ? FDT_STATUS_OKAY : FDT_STATUS_DISABLED;
 	nodeoff = fdt_set_status_by_compatible(blob, "edt,etml0700z9ndha", status);
diff --git a/include/configs/stm32mp25_common.h b/include/configs/stm32mp25_common.h
index 38c5be217b5..877879dc9d0 100644
--- a/include/configs/stm32mp25_common.h
+++ b/include/configs/stm32mp25_common.h
@@ -191,6 +191,7 @@
 		"fi;\0" \
 	"android_mmc_bootargs=" \
 		"env set bootargs ${android_bootargs} " \
+		   "${android_lcd_bootargs} " \
 		   "androidboot.serialno=${serial#} " \
 		   "androidboot.slot_suffix=_${suffix} " \
 		   "androidboot.force_normal_boot=${android_force_normal_boot} " \
-- 
2.34.1

